pkg/operator/ceph/cluster/cluster_cleanup.go

```diff
@@ -0,0 +1,145 @@
+	hostnameMap, err := k8sutil.GetNodeHostNames(c.context.Clientset)
+	if err != nil {
+		logger.Infof("error finding hostname. %+v", err)
+		return
+	}
+	for _, hostname := range hostnameMap {
+		storageNode := rookv1.Node{
+			Name: hostname,
+		}
+		validStorage.Nodes = append(validStorage.Nodes, storageNode)
+	}
+
+	validNodes := k8sutil.GetValidNodes(validStorage, c.context.Clientset, cephv1.GetOSDPlacement(cluster.Spec.Placement))
+
+	for i, node := range validNodes {
+		logger.Infof("starting clean up job on node %q", node.Name)
+		jobName := k8sutil.TruncateNodeName("cluster-cleanup-job-%s", strconv.Itoa(i))
+		envVars := []v1.EnvVar{
+			{Name: dataDirHostPath, Value: fmt.Sprintf("%s/%s", cluster.Spec.DataDirHostPath, cluster.Namespace)},
+			{Name: cleanDataDirHostPath, Value: "true"},
+		}
+		volumes := []v1.Volume{}
+		volumeMounts := []v1.VolumeMount{}
+		myVolume := v1.Volume{Name: "cleanup-volume", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: cluster.Spec.DataDirHostPath}}}
+		volumes = append(volumes, myVolume)
+		myVolumeMount := v1.VolumeMount{Name: "cleanup-volume", MountPath: cluster.Spec.DataDirHostPath}
+		volumeMounts = append(volumeMounts, myVolumeMount)
+		podSpec := v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: jobName,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:  jobName,
+					Image: c.rookImage,
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &privileged,
+					},
+					VolumeMounts: volumeMounts,
+					Env:          envVars,
+					Args:         []string{"ceph", "clean"}}},
+
+				Volumes:       volumes,
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
+		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: node.Name}
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      jobName,
+				Namespace: cluster.Namespace,
+				Labels:    controller.AppLabels(appName, cluster.Namespace),
+			},
+			Spec: batch.JobSpec{
+				Template: podSpec,
+			},
+		}
+
+		_, err := c.context.Clientset.BatchV1().Jobs(cluster.Namespace).Create(job)
+		if err != nil {
+			logger.Error("error running cluster clean up job on node %q. %+v", node.Name, err)
+		}
+
+	}
+
+}
+
+func (c *ClusterController) waitForCephDaemonCleanUp(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
+	for {
+		monDeps, err := k8sutil.GetDeployments(c.context.Clientset, cluster.Namespace, "app=rook-ceph-mon")
+		if err != nil {
+			return errors.Wrapf(err, "failed to list MON deployments")
+		}
+		osdDeps, _ := k8sutil.GetDeployments(c.context.Clientset, cluster.Namespace, "app=rook-ceph-osd")
+		if err != nil {
+			return errors.Wrapf(err, "failed to list OSD deployments")
+		}
+		if len(monDeps.Items) == 0 && len(osdDeps.Items) == 0 {
+			logger.Infof("ceph daemons cleaned up")
+			break
+		}
+
+		logger.Infof("waiting for ceph daemons in cluster %s to be cleaned up. Retrying in %s",
```

Comment:
How about logging how many of each type of daemon we are waiting to stop? We already have the deployments, so we could log the count of each one in this message.

Reply:
Updated log message: 
```
2020-03-13 13:01:44.727746 I | op-cluster: waiting for ceph daemons in cluster rook-ceph to be cleaned up. Retrying in 5s
2020-03-13 13:01:44.727779 I | op-cluster: hostname "ip-10-0-165-255". Pending ceph daemon count: 1
2020-03-13 13:01:44.727788 I | op-cluster: hostname "ip-10-0-158-135". Pending ceph daemon count: 2
2020-03-13 13:01:44.727796 I | op-cluster: hostname "ip-10-0-139-226". Pending ceph daemon count: 1
2020-03-13 13:01:49.781522 I | op-cluster: waiting for ceph daemons in cluster rook-ceph to be cleaned up. Retrying in 5s
2020-03-13 13:01:49.781554 I | op-cluster: hostname "ip-10-0-165-255". Pending ceph daemon count: 1
2020-03-13 13:01:54.817837 I | op-cluster: all ceph daemons cleaned up
2020-03-13 13:01:54.817890 I | op-cluster: starting clean up job on node "ip-10-0-165-255"
```

@travisn  please let me know if this looks good.

Reply:
I'd suggest we show how many of each daemon. I'm thinking like this:
```
waiting for ceph daemons (rook-ceph-mon: 3, rook-ceph-osd: 10, rook-ceph-mds: 1) in cluster "rook-ceph" to be stopped. 
```
In the `for` loop on line 155 you could append to that list of remaining pods, then after the `for` loop is done on line 164 you could write that log entry.

Reply:
Updated the verbiage a bit. Because we already have a different log that says we are waiting for <n> seconds and that's in a different method. 
 
```existing ceph daemons in the namespace "rook-ceph" - rook-ceph-mon: 2, rook-ceph-osd: 1, rook-ceph-mds: 0, rook-ceph-rgw: 0, rook-ceph-mgr: 1, rook-ceph-rbd-mirror: 0```